### ISSUE TITLE
Specify that DFC LDP servers must support PATCH and/or PUT

### ISSUE THREAD

**Issue by lecoqlibre** [datafoodconsortium/standard#7]
@simonLouvet We should specify that DFC LDP servers must support PATCH and/or PUT as the LDP spec does not require servers  to support them:
> 4.2.4 HTTP PUT
>
> Per [[RFC7231](https://www.w3.org/TR/ldp/#bib-RFC7231)], this HTTP method is optional and this specification does not require [LDP > servers](https://www.w3.org/TR/ldp/#dfn-ldp-server) to support it.

> 4.2.7 HTTP PATCH
>
>Per [[RFC5789](https://www.w3.org/TR/ldp/#bib-RFC5789)], this HTTP method is optional and this specification does not require [LDP >servers](https://www.w3.org/TR/ldp/#dfn-ldp-server) to support it.

**Comment by simonLouvet** [comment]
exact. Put used because patch  involves a great deal of complexity

**Comment by RachL** [comment]
@simonLouvet @lecoqlibre can we close this one?

**Comment by simonLouvet** [comment]
@RachL No, question of @lecoqlibre don't have explicit  answer.

I think DFC servers ave to implement PUT because it's the easiest way to update data on one platform from another. On the other hand, we can rule out PATCH, which is complex to specify. Is ok for you @lecoqlibre ?

**Comment by lecoqlibre** [comment]
I would like the standard to have a possibility to patch data. PATCH is very useful as it allows to change only a part of the data. Also it consumes less resources as we only pass the modified data and not the whole dataset.I was thinking about using N3 patch like in the Solid protocol. Why do you think it's complex to specify @simonLouvet?

**Comment by lecoqlibre** [comment]
Any news @simonLouvet?

**Comment by simonLouvet** [comment]
I don't think N3 patch can be easily implemented by platforms that don't manage a semantic database. They have to interpret the N3 patch and translate it into their platform's storage system.

I still don't think it's a good idea to impose it. If it is allowed (but not required), I don't know what it will change. Backward compatibility will have to be maintained to manage platforms that don't implement it for a marginal gain in performance.

Translated with DeepL.com (free version)

**Comment by lecoqlibre** [comment]
I agree that PATCH requests are complicated even more when blank nodes are involved (because we can't update something unidentified) although it would be nice to have a way to partially update resources. [Linked Data Patch Format](https://www.w3.org/TR/ldpatch/) might be a solution for the future.

The gain is marginal for few resources but when you want to update many resources it becomes signifiant.

I'm OK with the fact that we don't impose anything on PATCH requests. Platforms can implement their own way of patching resource. 

Without a standardized patching mechanism a platform won't be able to patch a resource on an other platform unless they are using the same mechanism.

The remaining todo is then:
- [ ] Specify that DFC LDP servers must support PUT requests

**Comment by lecoqlibre** [comment]
In fact beside the gain in performance, PUT is dangerous. Data can be lost if the request is not well formed. I think we should support PATCH to enforce data coherence. We can state that we do not support blank nodes so it will ease the N3 Patch implementation. Translating a N3 Patch into the platform storage system even without a quadstore is feasible.

**Comment by simonLouvet** [comment]
I am not opposed to the concept. Once again, platforms are part of governance and their opinion must be sought.
